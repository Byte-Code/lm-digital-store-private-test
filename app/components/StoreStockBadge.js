import React, { PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';
import CheckIcon from 'material-ui/svg-icons/navigation/check';
import BlockIcon from 'material-ui/svg-icons/navigation/close';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Availability = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 10px;
  &>p {
    font-size: 16px;
    &:first-child {
      font-family: LeroyMerlinSans Bold;
    }
  }
`;

function getStockLabel(stock, stockStatus) {
  if (stock > 0) {
    return stock + (stock === 1 ? ' prodotto disponibile' : ' prodotti disponibili');
  }
  switch (stockStatus) {
    case 'notAvailable':
      return 'Prodotto non disponibile';
    case 'availableOnOrder':
      return 'Prodotto disponibile su ordinazione';
    case 'onRestock':
    default:
      return 'Prodotto in riassortimento';
  }
}

function getStockIcon(stock, stockStatus) {
  if (stock > 0) {
    return <CheckIcon style={{ width: 40, height: 45 }} color="#58c527" />;
  }
  if (stockStatus === 'notAvailable') {
    return <BlockIcon style={{ width: 65, height: 45 }} color="#dddddd" />;
  }
  return null;
}

const StoreStockBadge = ({ currentStoreStock, storeName, showStore }) => {
  const availability = currentStoreStock.get('storeStock');
  const stockStatus = currentStoreStock.get('stockStatus');
  const label = getStockLabel(availability, stockStatus);
  const icon = getStockIcon(availability, stockStatus);

  return (
    <Wrapper>
      {icon}
      <Availability>
        <p>{label}</p>
        {showStore && (<p>{`in negozio a ${storeName}`}</p>)}
      </Availability>
    </Wrapper>
  );
};

// TODO find a better solution for hideStore
StoreStockBadge.propTypes = {
  currentStoreStock: ImmutablePropTypes.map.isRequired,
  storeName: PropTypes.string.isRequired,
  showStore: PropTypes.bool
};

StoreStockBadge.defaultProps = {
  showStore: true
};

export default StoreStockBadge;
