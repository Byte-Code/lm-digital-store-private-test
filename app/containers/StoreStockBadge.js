import { connect } from 'react-redux';

import StoreStockBadge from '../components/StoreStockBadge';
import { getStoreName } from '../reducers/selectors';

const mapStateToProps = (state) => ({
  storeName: getStoreName(state)
});

export default connect(mapStateToProps, {})(StoreStockBadge);
