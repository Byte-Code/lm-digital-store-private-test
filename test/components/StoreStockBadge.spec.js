import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import StoreStockBadge from '../../app/components/StoreStockBadge';

const storeName = 'storeName';
const currentStoreStock = fromJS({
  storeStock: 10,
});

// TODO finish here
it('should render properly', () => {
  const result = shallow(
    <StoreStockBadge
      storeName={storeName}
      currentStoreStock={currentStoreStock}
      showStore
    />
  );
  expect(result).toMatchSnapshot();
});
