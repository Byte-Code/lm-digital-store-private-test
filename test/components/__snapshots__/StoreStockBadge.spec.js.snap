// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<styled.div>
  <NavigationCheck
    color="#58c527"
    style={
      Object {
        "height": 45,
        "width": 40,
      }
    }
  />
  <styled.div>
    <p>
      10 prodotti disponibili
    </p>
    <p>
      in negozio a storeName
    </p>
  </styled.div>
</styled.div>
`;
